book/website/reproducible-research/rdm/rdm-spreadsheets.md

```diff
@@ -1,35 +1,134 @@
+|2. Each observation must have its own row.|
+|3. Each value must have its own cell.  |
+
+There are data validation tools available, like https://goodtables.io, that allows you check automaticall whether your spreadsheets are tidy.
+
+## 3. Other tips
+
+### Deal with time information
+
+While dates should be written as yyyy-mm-dd, excel and other softwares tend to transform this data into their own format for dates (even during data import from a .csv file !). The only 100% secure way to deal with this is to make different columns for years, months and day and recreate the date in the software used for analysis. Time entered with hh:mm:ss normally works.
+
+### Working with several sheets
+
+You often use several sheets for different but related data. It is a very useful tool indeed, expecially when one wants to share the whole data with colleagues. .csv files on the other hand are only saving one sheet at a time. In addition, most data analysis software have ways to import .xlsx files, such that the format is qutie a standard. The practical solution is to work with the .xlsx format, while making sure all the information is avaialble in a .csv export for each sheet. A better solution, especially for long term storage, is to save all sheets separately in a csv file and zip them together. This latter solution allows to contain also extra documenations that could be in a different format (for example a text file explaning the meaning of the headers and the unit chosen)
+
+### Spreadsheet design
+
+In order to avoid mistakes and be most efficient, the same spreadsheet should be used to collect the data (even if this happens on a printed version of the spreadsheet) and analyse the data. The poses some design questions, especially for information which are unique for one experiment but that needs to be included in the data, as it changes between experiments (experimentator, temperature of the room,...). You indeed want that information into one column, but you would like to enter it only once (especially on a printed version). One solution is to move these columns on a second (Non-printed) page on the same spreadsheet, and playing with the headers and footers to enter the information on paper. 
+
+The way you enter the information (that is the way you design your headers and cell content) may be different depending on the analysis you want to perform later. One should still always try to be as generic and objective as possible, and think about additional analysis one may want to perform. As an example, let's suppose you are interested in a genetic tree of people with blue eyes. While you may have a column "has_blue_eyes" with the value true or false, it would be better to have a column about the eye colour (brown - blue- green) as it will be easy to create the  "has_blue_eyes" information from it, and allow additional analysis. (In this case, blue and green eyes might be the same gene for eyes, with a different skin color).
```

Comment:
I like the idea to use the iris dataset, but we would need an example of new data entries (otherwise we are not really using spreadsheet). I will think about how the dataset could have been collected in a bad way, and give the iris dataset as the good way. would this be better?

```
The way you enter the information (that is the way you design your headers and cell content) may be different depending on the analysis you want to perform later. One should still always try to be as generic and objective as possible, and think about additional analysis one may want to perform. 

As an example, let's suppose you are interested in depicting if the number of iris flowers whose sepal length is longer than 6 cm is different in three species. You may be inclined to record a column recording is-sepal-longer-than-6cm, but this will restrict the analysis you may perform. A better solution is to record the length of the sepal, and create automatically the categorization later.

If you are using R, you would then plot what you wanted with:
iris %>% dplyr::mutate ("is-sepal-longer-than-6cm" = ifelse(Sepal.Length >6, TRUE, FALSE)) %>%
  ggplot2::ggplot (aes (x=`is-sepal-longer-than-6cm` , fill= Species)) + ggplot2::geom_bar()

```

Reply:
I like this a lot!